### ISSUE TITLE
Add context menu to the added feature to recompute

### ISSUE THREAD

**Issue by FlorianPommerening** [FlorianPommerening/FingerJoints#3]
This is a somewhat messy work-around to the fact that plugins cannot react to changes in the model. It is an imperfect solution and a lot of work because it would also require to store and recover all parameters in some way. I'm not sure I'll do this but want to keep the idea in mind.

**Comment by FlorianPommerening** [comment]
https://forums.autodesk.com/t5/fusion-360-api-and-scripts/help-with-add-in-for-cutting-finger-joints/m-p/9514508

**Comment by it-ony** [comment]
Does a custom feature also help out here: 
http://help.autodesk.com/view/fusion360/ENU/?guid=GUID-FA7EF128-1DE0-4115-89A3-795551E2DEF2

I didn't check out the code yet, but after I played around with the addin, I realized, that changing the source bodies won't reflect in changed finger joins.

**Comment by FlorianPommerening** [comment]
Yes, that seems like the perfect solution for this problem. Too bad it is still in preview mode but I might try it out in a branch.

**Comment by it-ony** [comment]
Let me know, how I can participate. I'll check out the code tonight.

**Comment by FlorianPommerening** [comment]
I wanted to look into this today but my educational license expired and my employee ID apparently is not sufficient to prove my educational status.

Anyway, as long as custom features are marked as a preview feature, Autodesk recommends against publishing code that uses them because the API interface is not stable yet. So even if there is a solution, it'll have to wait until custom features make it out of preview mode.

**Comment by it-ony** [comment]
Sounds correct. My license is expiring soonish as well. Please let me know, if I can help out here.

**Comment by Fardenco** [comment]
Too bad it's still experimental, being able to modify the feature afterward as we always do with "stock" features would make your tool so much easier to use !
Anyway, thanks for your awesome work, I'll follow this closely.

**Comment by FlorianPommerening** [comment]
I completely agree @Fardenco. When I started with this project, I just assumed this would be part of the API and was disappointed to find that it wasn't possible to recompute a feature at all (Custom Features didn't exist yet). Now at least we have the preview mode. Let's hope they finalize it soon.

By the way, I changed the issue title to reflect that it is now about implementing the add-in as a custom feature. This doesn't change anything, we still need to wait for the API to stabilize but the old title was about a workaround that would have been the only choice without custom features.

**Comment by it-ony** [comment]
I've seen that https://github.com/FlorianPommerening/FingerJoints/issues/10 is closed, which was a requirement to get this working. Can we make use of the API in preview mode at least on a feature branch? I really have pain, removing the finger joints and recreate them, every time I change a dimension.

**Comment by it-ony** [comment]
I also wonder, if instead of creating a base feature, we could create a sketch with projected geometry to create the fingers. The sketch would re-compute if dimensions change. This will only work for a fixed amount of fingers, but it would be a start as well. The custom feature seems anyway, the right way to go.

**Comment by Fardenco** [comment]
I think the priority should be on Autodesk's side, they must finally finish the custom features system and make it available for everyone
It's been a while and nothing seems to be moving

**Comment by it-ony** [comment]
I assume it's a chicken egg problem. It's not moving as nobody gives feedback and start using the API. I'll give it a try within the next days as I'm always having a lot of trouble when my design changes.

**Comment by FlorianPommerening** [comment]
I thought so too but it doesn't seem to be. Looking at the end of the discussion thread for the preview feature, they have lots of feedback but don't get around to working on it. Or rather, the solution they are planning has to fit with their overall design and this probably requires some other changes first (see message 61 in the preview discussion thread). 

I already tried an implementation in a branch but ran into problems and reported them in the thread. Since then I don't think the interface changed, so the ball is in their corner now.

But honestly, even if I could make progress on this now, my day job currently takes a lot of my time and it would be tricky finding extra time to work on the add in.

**Comment by it-ony** [comment]
Got it. Thanks @FlorianPommerening. If the branch pushed?

**Comment by FlorianPommerening** [comment]
It is a bit out-dated by now but yes, what I have is in `experimental-custom-feature`.

**Comment by it-ony** [comment]
Got your branch half working. A PR #17 has been created. It's not really updating the finger joints yet and I'm wondering, if it's possible to remove all the base features from the CustomFeature and just create new ones.

**Comment by it-ony** [comment]
https://github.com/FlorianPommerening/FingerJoints/pull/30
